packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -300,14 +302,20 @@ abstract class Artifacts {
   /// all artifacts.
   ///
   /// If a [fileSystem] is not provided, creates a new [MemoryFileSystem] instance.
-  ///
-  /// Creates a [LocalEngineArtifacts] if `localEngine` is non-null
-  factory Artifacts.test({String? localEngine, FileSystem? fileSystem}) {
-    fileSystem ??= MemoryFileSystem.test();
-    if (localEngine != null) {
-      return _TestLocalEngine(localEngine, fileSystem);
-    }
-    return _TestArtifacts(fileSystem);
+  factory Artifacts.test({FileSystem? fileSystem}) {
+    return _TestArtifacts(fileSystem ?? MemoryFileSystem.test());
+  }
+
+  /// A test-specific implementation of artifacts that returns stable paths for
+  /// all artifacts, and uses a local engine.
+  /// 
+  /// If a [fileSystem] is not provided, creates a new [MemoryFileSystem] instance.
+  factory Artifacts.testLocalEngine({
```

Comment:
Should this have a @visibleForTesting annotation if it is only used for testing?

Reply:
Done (and applied above).